torch/distributed/fsdp/_state_dict_utils.py

```diff
@@ -730,21 +730,24 @@ def _post_state_dict_hook(
         for key, tensor in sorted(processed_state_dict.items()):
             if key.startswith(prefix) and isinstance(tensor, torch.Tensor):
                 local_shape = tensor.shape
+                device = None
```

Comment:
Hello, is there a chance `isinstance(tensor, ShardedTensor)` and `isinstance(tensor, DTensor)` both `False`? Then `device` is None in line 750. How about assign `device = tensor.device` in here(since it's default value), and reassign in following if statement. :D

Reply:
@hippocookie  : thanks for the suggestion, will incorporate that

Reply:
@hippocookie  : Initializing device with tensor.device instead of None , before checking if its a sharded tensor will still land up in the same problem, since tensor.device can be a cuda tensor.